android/engine/src/test/resources/extractions/record-all-immunization/structure-map.txt

```diff
@@ -0,0 +1,39 @@
+map "http://fhir.labs.smartregister.org/fhir/StructureMap/followup-structuremap" = 'ImmunizationCarePlan'
+
+uses "http://hl7.org/fhir/StructureDefinition/QuestionnaireResponse" as source
+uses "http://hl7.org/fhir/StructureDefinition/Bundle" as target
+
+group ImmunizationCarePlan(source src : QuestionnaireResponse, target tgt: Bundle) {
+            src -> create('Encounter') as encounter
+            then ExtractEncounter(src, encounter, tgt),
+                 ExtractImmunization(src, encounter, tgt) "r_careplan";
+}
+
+group ExtractEncounter(source src : QuestionnaireResponse, target encounter: Encounter, target bundle : Bundle) {
+        src -> encounter.id = uuid(),
+            encounter.status = "finished",
+            encounter.serviceType = cc("http://terminology.hl7.org/CodeSystem/service-type", "57", "Immunization") "rule_ee_4";
+
+        src -> bundle.entry as entry, entry.resource = encounter "r_enc_cp";
+}
```

Comment:
If this is to fulfil the following discussion then the encounters generated are not correct. https://github.com/opensrp/fhircore/discussions/2142 We need 2 Encounters + 1 Immunization resources

Reply:
Also, the encounters do not have all the required fields.

Reply:
@dubdabasoduba This was to test and add a base for extraction in record all and was done as part of https://github.com/opensrp/fhir-resources/issues/1006 ticket. We can add corresponding entities as per requirement. Kindly share the ticket number if any is specific for this. This looks like FhirCareplanGenerateTest but it is specifically to add / test extraction of questionnaires which is different from CarePlan generation.